bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/discovery/MDNSDiscoveryService.java

```diff
@@ -68,59 +73,41 @@ public void run() {
                 startScan();
             }
         }, 0, TimeUnit.SECONDS);
-        for (MDNSDiscoveryParticipant participant : participants) {
-            mdnsClient.getClient().addServiceListener(participant.getServiceType(), this);
-        }
-    }
-
-    @Override
-    protected void stopBackgroundDiscovery() {
-        for (MDNSDiscoveryParticipant participant : participants) {
```

Comment:
I explained why in issue #1745.
One listener is already registered in setMDNSClient (when binding the service) and always active. Managing a second listener has no interest, methods are called twice during the discovery.